examples/stateful-precompile/src/main.rs

```diff
@@ -63,58 +63,41 @@ pub struct MyEvmFactory {
 }
 
 impl EvmFactory<EvmEnv> for MyEvmFactory {
-    type Evm<DB: Database, I> = EthEvm<DB, I>;
+    type Evm<DB: Database, I: Inspector<EthEvmContext<DB>, EthInterpreter>> = WrappedEthEvm<DB, I>;
     type Tx = TxEnv;
     type Error<DBError: core::error::Error + Send + Sync + 'static> = EVMError<DBError>;
     type HaltReason = HaltReason;
     type Context<DB: Database> = EthEvmContext<DB>;
 
-    fn create_evm<'a, DB: Database + 'a>(
-        &self,
-        db: DB,
-        input: EvmEnv,
-    ) -> Self::Evm<DB, NoOpInspector> {
-        let cfg_env_with_handler_cfg = CfgEnvWithHandlerCfg {
-            cfg_env: input.cfg_env,
-            handler_cfg: HandlerCfg::new(input.spec),
-        };
-
+    fn create_evm<DB: Database>(&self, db: DB, input: EvmEnv) -> Self::Evm<DB, NoOpInspector> {
```

Comment:
UX wise it would be a lot simpler if we didn't have to deal with inspectors on this level, then we could also get rid of create_evm_with_inspector function here entirely.
not a blocker, but it illustrates why dealing with inspectors on this level feels strange

Reply:
This feels possible and look like a proper way to do things, but needs to be cleared in Revm.

Reply:
Have added a issue here: https://github.com/bluealloy/revm/issues/2069 it shouldn't take a lot of time and if it is quick fix i will do it tomorrow